src/gateways/googleAnalyticsClient.ts

```diff
@@ -28,7 +28,14 @@ interface SetCommand {
   };
 }
 
-type Command = SetCommand | SendCommand;
+interface SetCommandCustomDimension {
+  name: 'set';
+  payload: {
+    dimension3: string | undefined;
```

Comment:
you can just change SetCommand so that `userId` and `dimensionId` are optional like: 

```
interface SetCommand {
  name: 'set';
  payload: {
    userId?: string | undefined;
    dimension3?: string | undefined;
  };
}
```

i think the way you have it will work, but the pattern we're following here with the `Command` type is a [discriminated-union](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) which is easier to follow if there is a top level discriminator key (`name` in this case).